File: SpriteKitLib/src/main/java/br/com/insanitech/spritekit/SKRect.java
package br.com.insanitech.spritekit;

import br.com.insanitech.spritekit.opengl.model.GLPoint;
import br.com.insanitech.spritekit.opengl.model.GLRect;

/**
 * Created by anderson on 7/4/15.
 */

public class SKRect extends GLRect {
    public SKRect() {
        super();
    }

    public SKRect(float x, float y, float width, float height) {
        super(x, y, width, height);
    }

    public boolean containsPoint(SKPoint point) {
        return super.containsPoint(point);
    }

    @Override
    public String toString() {
        return "{ x: " + getX() + ", y: " + getY() + ", w: " + getWidth() + ", h: " + getHeight() + " }";
    }
}

File: SpriteKitLib/src/main/java/br/com/insanitech/spritekit/SKRect.kt
package br.com.insanitech.spritekit

import br.com.insanitech.spritekit.opengl.model.GLRect

/**
 * Created by anderson on 7/4/15.
 */

class SKRect : GLRect {
    constructor() : super() {}

    constructor(x: Float, y: Float, width: Float, height: Float) : super(x, y, width, height) {}

    constructor(other: SKRect) : super(other.x, other.y, other.width, other.height) {}

    fun containsPoint(point: SKPoint): Boolean {
        return super.containsPoint(point)
    }

    override fun toString(): String {
        return "{ x: $x, y: $y, w: $width, h: $height }"
    }
}